### PR TITLE
Clean up network device discovery replies due to mechanism update

### PR DESCRIPTION
**Pull request by t-mon**
nymea-plugins-modbus pull request checklist:

- [x] Make sure the pull request's title is of format "Plugin name: Add support for xyz" or "New plugin: Plugin name"

- [x] Did you test the changes on hardware, if not (e.g. absence of required hardware), please mention a person to confirm it has been tested.

- [x] Did you update the plugin's README.md accordingly?

- [x] Did you update translations (`cd builddir && make lupdate`)?

### CODE REVIEW

File: huawei/huaweifusionsolardiscovery.cpp [hunk 1]
```diff
@@ -46,15 +46,11 @@ void HuaweiFusionSolarDiscovery::startDiscovery()
     qCInfo(dcHuawei()) << "Discovery: Start searching for Huawei FusionSolar SmartDongle in the network...";
     NetworkDeviceDiscoveryReply *discoveryReply = m_networkDeviceDiscovery->discover();
 
-    // Check any already discovered infos..
-    foreach (const NetworkDeviceInfo &networkDeviceInfo, discoveryReply->networkDeviceInfos()) {
-        checkNetworkDevice(networkDeviceInfo);
-    }
-
```
Comment (mzanetti) [inline huawei/huaweifusionsolardiscovery.cpp:53]: given this is not needed any more, you can also remove the entire m_verifiedNetworkDevices member in all the plugins...

File: huawei/huaweifusionsolardiscovery.cpp [hunk 2]
```diff
@@ -46,15 +46,11 @@ void HuaweiFusionSolarDiscovery::startDiscovery()
...
     // Immedialty check any new device gets discovered
```
Comment (mzanetti) [inline huawei/huaweifusionsolardiscovery.cpp:49]: this comment doesn't make that much sense any more now.

File: huawei/huaweifusionsolardiscovery.cpp [hunk 3]
```diff
@@ -46,15 +46,11 @@ void HuaweiFusionSolarDiscovery::startDiscovery()
...
     connect(discoveryReply, &NetworkDeviceDiscoveryReply::networkDeviceInfoAdded, this, &HuaweiFusionSolarDiscovery::checkNetworkDevice);
 
     // Check what might be left on finished
```
Comment (mzanetti) [inline huawei/huaweifusionsolardiscovery.cpp:52]: also this comment is wrong nowadays

File: huawei/huaweifusionsolardiscovery.cpp (hunk 1 above)
Comment (mzanetti) [reply huawei/huaweifusionsolardiscovery.cpp:53]: also this whole block in finished() can be removed
```
        // Send a report request to nework device info not sent already...
        foreach (const NetworkDeviceInfo &networkDeviceInfo, m_networkDeviceInfos) {
            if (!m_verifiedNetworkDeviceInfos.contains(networkDeviceInfo)) {
                checkNetworkDevice(networkDeviceInfo);
            }
        }
```
In fact, I think the m_networkDeviceInfos member variable is not needed any more either for the majority of the plugins (depends a bit how the post-processing looks like)

**Review by mzanetti — CHANGES_REQUESTED** [review]

File: huawei/huaweifusionsolardiscovery.cpp (hunk 3 above)
Comment (t-mon) [reply huawei/huaweifusionsolardiscovery.cpp]: done

File: huawei/huaweifusionsolardiscovery.cpp (hunk 1 above)
Comment (mzanetti) [reply huawei/huaweifusionsolardiscovery.cpp:53]: what about this?

Comment (t-mon) [reply huawei/huaweifusionsolardiscovery.cpp:53]: should all be fixed, see the diff underneeth

**Pull request merged**